### DISCUSSION TITLE
How to customise revealjs slide title formatting

### DISCUSSION THREAD

**Discussion by jpmorr** [quarto-dev/quarto-cli#9188] [Q&A] [answered]
### Description

I'm trying to create a `revealjs` theme that resembles some of the standard powerpoint/keynote presentations that have a solid colour under the title across the full width of the slide, much like the image below.

![image](https://github.com/quarto-dev/quarto-cli/assets/10863809/ccc9d5ee-1359-4cb8-afb8-68beb788379a)

I'm trying to make my quarto slides fit with our standard corporate theme which resembles the above slide and then add a logo in the top right corner.

I added the following code to my `custom.scss` thinking it would be as simple as that but the outcome is slightly different - see below:
```
.reveal h2 {
    background-color: $navy_blue;
    padding: 10px 0px 10px 20px;
    height: 85px;
    color: $golden
    border-radius: 0px;
}
```

![image](https://github.com/quarto-dev/quarto-cli/assets/10863809/9aff3fdb-2bab-49b2-9c8e-fa7c019086ef)

It appears that there is some larger canvas outside the `slides` area because it looks correct in the slides overview.  I guess because it's a larger canvas it has to be a single uniform colour across the screen? I don't immediately see a way to achieve this simple style.

I set the slide aspect ratio (height & width)  to the same aspect ratio as the display, thinking that would stop this happening, bu tthat extra white border all around still exists.

**Comment by mcanouil** [comment]
There are quite a few discussions regarding customisation of Reveal.js, I suggest you look for them.

- https://github.com/quarto-dev/quarto-cli/discussions?discussions_q=is%3Aopen+label%3Arevealjs+label%3Athemes

Additionally/alternatively, when working with HTML/CSS, you need to use your favourite browser inspect/developer mode to understand the CSS/HTML structure, then create your (S)CSS rules accordingly.


PS: sharing screenshots instead of code is not advisable as we can't use screenshots but we can use code.

**Comment by cderv** [comment]
Revealjs is quite specific in how in handles background. You need to look at the Revealjs documentation. 

You can just set the background on an element because backgrounds are decoupled from content (by Revealjs itself) - you need to target both. 

I don't know to which extent Revealjs will allow to have such layout as a title banner - I would suggest looking at example of revealjs presentation online, and its theme. If Revealjs can't do, unfortunately Quarto won't be able to do it either. 

We need certainly to improve the theming mechanism for revealjs, but layout of the slides are quite different. 

In the past, we've done such background change by creating some image content used as background onto which the content is then organized and styled. (so it would be here a background image with a color banner and white background below). 

Hope it helps

**Comment by jpmorr** [comment, marked as answer]
Actually, I've had a chance to look at this again over the last few days and the important part was to remove the margin that is set by default in `revealjs`. With `margin: 0.0` in my settings the slide now takes up the full screen as expected and the `h2` formatting is all that is required to achieve the desired look (adjusting `h2` `padding` as required).

The remaining issue is that the slide number now gets consumed by the title bar so I need to move that to a new location which I believe shouldn't be too difficult either.

**Reply by jpmorr** [reply]
Just to note, after putting the presentation together, I've realised that an extra step is necessary if `margin` is reduced to zero - all slide items such as paragraphs, lists, columns, divs, etc require some left padding to move them away from the screen edge. 

While it requires an extra bit of effort to do, it's necessary to achieve a slide that doesn't look weird because everything starts on the edge.
